### PR TITLE
installation: package.json addition

### PR DESCRIPTION
**Pull request by jmartinm**
* Adds package.json to simplify installation of npm packages.

* This allows as well to control what versions we install.

* Updates installation guide accordingly.

Signed-off-by: Javier Martin Montull <javier.martin.montull@cern.ch>

### CODE REVIEW

**Comment by jmartinm** [comment]
This was started because of problems on ``clean-css``.

Installing latest version causes problems as version 3 is quite different to version 2 (https://github.com/jakubpawlowicz/clean-css#how-to-upgrade-clean-css-from-2x-to-3x)

So the overlay css was not taking precedence over the Invenio one any more.

**Comment by jacquerie** [comment]
Another argument for having this is the (possible) future addition of the Karma test runner, which ships plugins for PhantomJS or Chrome integrations as Node modules.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "dependencies": {
+    "less": "*",
```
Comment (lnielsen) [inline package.json]: I think we should be using less 1.7.5

File: INSTALL.rst [hunk 2]
```diff
@@ -458,12 +458,9 @@ they are not in the environment ``$PATH`` already.
...
 
-    # Global installation
-    $ sudo su -c "npm install -g less clean-css requirejs uglify-js"
-
-    # or
-    # Local installation
-    (invenio)$ npm install less clean-css requirejs uglify-js
+    # Local installation (using package.json)
+    (invenio)$ cdvirtualenv src/invenio
+    (invenio)$ npm install
```
Comment (lnielsen) [inline INSTALL.rst]: perhaps ``-g``?

**Comment by lnielsen** [comment]
:+1:

File: INSTALL.rst (hunk 2 above)
Comment (jmartinm) [inline INSTALL.rst]: I don't think ``-g`` is supported from a ``package.json`` ? At least it did not work for me and could not find online how to do it.

http://stackoverflow.com/questions/6480549/install-dependencies-globally-and-locally-using-package-json

File: package.json (hunk 1 above)
Comment (jmartinm) [inline package.json]: Have you found problems with newer version?

**Pull request merged**